smartsim/_core/control/previewrenderer.py

```diff
@@ -100,6 +106,17 @@ def as_toggle(_eval_ctx: u.F, value: bool) -> str:
     return "On" if value else "Off"
 
 
+@pass_eval_context
+def get_ifname(_eval_ctx: u.F, value: t.List[str]) -> str:
+    return next((item for item in value if "ifname" in item), str).split("=")[-1]
+
+
+@pass_eval_context
+def get_dbtype(_eval_ctx: u.F, value: str) -> t.Any:
+    db_type, _ = value.split("/")[-1].split("-", 1)
```

Comment:
ditto here. if `value == ""` the splits are going to blow up?

Reply:
added check for empty string